### ISSUE TITLE
NPE on start-up

### ISSUE THREAD

**Issue by SlimeDog** [CitizensDev/Citizens2#2937]
### `/version` output

```txt
>version
[07:23:59] [Server thread/INFO]: This server is running CraftBukkit version 3608-Spigot-6198b5a-b5aa0be (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT)
[07:23:59] [Server thread/INFO]: You are running the latest version
```
or
```
> version
[07:15:54 INFO]: This server is running Paper version git-Paper-300 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 54daefa)
You are running the latest version
```
```


### `/version Citizens` output

```txt
>version citizens
[07:24:44] [Server thread/INFO]: Citizens version 2.0.30-SNAPSHOT (build 2773)
[07:24:44] [Server thread/INFO]: Website: https://www.citizensnpcs.co
[07:24:44] [Server thread/INFO]: Author: fullwall
```

### Log file

https://paste.denizenscript.com/View/103590

### config.yml

https://blackdog.straight8.com/drops/config.yml

### saves.yml

https://blackdog.straight8.com/drops/saves.yml

### Description

- Update to Citizens 2.30.0-2773
- Start server

```
[07:13:29] [Server thread/WARN]: [Citizens] Task #106 for Citizens v2.0.30-SNAPSHOT (build 2773) generated an exception
java.lang.NullPointerException: metadata
        at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:39) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.setRecvByteBufAllocator(DefaultChannelConfig.java:325) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:79) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at net.citizensnpcs.util.EmptyChannel.<init>(EmptyChannel.java:15) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.util.NMSImpl.initNetworkManager(NMSImpl.java:1969) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.network.EmptyNetworkManager.<init>(EmptyNetworkManager.java:14) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.initialise(EntityHumanNPC.java:329) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.<init>(EntityHumanNPC.java:109) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.HumanController.createEntity(HumanController.java:52) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.npc.AbstractEntityController.spawn(AbstractEntityController.java:48) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:278) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.load(CitizensNPC.java:171) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.api.npc.SimpleNPCDataStore.loadInto(SimpleNPCDataStore.java:59) ~[Citizens-2.0.30-b2773.jar:?]
        at net.citizensnpcs.Citizens$CitizensLoadTask.run(Citizens.java:569) ~[Citizens-2.0.30-b2773.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftTask.run(CraftTask.java:82) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:415) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1245) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:394) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1197) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1013) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:293) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3608-Spigot-6198b5a-b5aa0be]
        at java.lang.Thread.run(Thread.java:1589) [?:?]
```

**Comment by fullwall** [comment]
Can you try the newest build please?

**Comment by SlimeDog** [comment]
The error still exists with Citizens 2.0.30-b2774 on both Spigot 1.19.2-3608 and Paper 1.19.2-300.

```
[07:54:51] [Server thread/WARN]: [Citizens] Task #104 for Citizens v2.0.30-SNAPSHOT (build 2774) generated an exception
java.lang.NullPointerException: metadata
        at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:39) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.setRecvByteBufAllocator(DefaultChannelConfig.java:325) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:79) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at net.citizensnpcs.util.EmptyChannel.<init>(EmptyChannel.java:15) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.util.NMSImpl.initNetworkManager(NMSImpl.java:1969) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.network.EmptyNetworkManager.<init>(EmptyNetworkManager.java:14) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.initialise(EntityHumanNPC.java:329) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.<init>(EntityHumanNPC.java:109) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.HumanController.createEntity(HumanController.java:52) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.npc.AbstractEntityController.spawn(AbstractEntityController.java:48) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:278) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.load(CitizensNPC.java:171) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.api.npc.SimpleNPCDataStore.loadInto(SimpleNPCDataStore.java:59) ~[Citizens-2.0.30-b2774.jar:?]
        at net.citizensnpcs.Citizens$CitizensLoadTask.run(Citizens.java:569) ~[Citizens-2.0.30-b2774.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.19.2.jar:git-Paper-300]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483) ~[paper-1.19.2.jar:git-Paper-300]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1114) ~[paper-1.19.2.jar:git-Paper-300]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:305) ~[paper-1.19.2.jar:git-Paper-300]
        at java.lang.Thread.run(Thread.java:1589) ~[?:?]
```

**Comment by mcmonkey4eva** [comment]
On Citizens-2775 and Paper-301, I don't get the same errors at all.
It might relate to one of the other plugins you have installed, or your Java version?
Are you able to test whether it replicates without other plugins on the server?

**Comment by SlimeDog** [comment]
I usually run Java 19, but the issue occurs with Java 17.0.4 as well. The issue occurs with Paper-301 and Citizens starting with 2770 (ie., it does not occur with Citizens-2769) and the same server setup, including all other plugins and configurations.

**Comment by mcmonkey4eva** [comment]
That's good to know, but, still testing whether it occurs without other plugins will be valuable

**Comment by SlimeDog** [comment]
If I remove enough of my 160 plugins (remember I run a plugin test server), it does not occur. It will take some time to isolate the conflict.

**Comment by SlimeDog** [comment]
It looks like the conflict is with [HeadsPluginAPI](https://github.com/CC007/HeadsPluginAPI/releases), which hasn't changed for months and did not cause issues prior to Citizens-2770. But if I remove it (and the two plugins that depend on it), the Citizens NPE does not occur. When included, HeadsPluginAPI produces no errors. I hope that is helpful.

**Comment by fullwall** [comment]
Hmm, the error is caused by having a different Netty version. Are you able to give me the output of Netty's Version.identify() with HeadsPluginAPI enable?

**Comment by SlimeDog** [comment]
How can I do that?

I can tell that in `./libraries/netty/netty-*/` there are various versions of netty components:
```
4.1.49.Final
4.1.74.Final
4.1.77.Final
```
Does that help?

**Comment by fullwall** [comment]
I've made it down to Netty 4.1.1 to see if that helps? Try build #2785

**Comment by SlimeDog** [comment]
```
[09:13:08] [Server thread/WARN]: [Citizens] Task #107 for Citizens v2.0.30-SNAPSHOT (build 2785) generated an exception
java.lang.NullPointerException: metadata
        at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:39) ~[netty-common-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.setRecvByteBufAllocator(DefaultChannelConfig.java:325) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:79) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75) ~[netty-transport-4.1.77.Final.jar:4.1.77.Final]
        at net.citizensnpcs.util.EmptyChannel.<init>(EmptyChannel.java:15) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.util.NMSImpl.initNetworkManager(NMSImpl.java:2039) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.network.EmptyNetworkManager.<init>(EmptyNetworkManager.java:14) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.initialise(EntityHumanNPC.java:329) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.EntityHumanNPC.<init>(EntityHumanNPC.java:109) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.nms.v1_19_R1.entity.HumanController.createEntity(HumanController.java:52) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.npc.AbstractEntityController.spawn(AbstractEntityController.java:48) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.spawn(CitizensNPC.java:277) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.npc.CitizensNPC.load(CitizensNPC.java:170) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.api.npc.SimpleNPCDataStore.loadInto(SimpleNPCDataStore.java:59) ~[Citizens-2.0.30-b2785.jar:?]
        at net.citizensnpcs.Citizens$CitizensLoadTask.run(Citizens.java:578) ~[Citizens-2.0.30-b2785.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftTask.run(CraftTask.java:82) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at org.bukkit.craftbukkit.v1_19_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:415) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1245) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:394) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1197) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at net.minecraft.server.MinecraftServer.v(MinecraftServer.java:1013) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:293) ~[spigot-1.19.2-R0.1-SNAPSHOT.jar:3609-Spigot-6198b5a-f3dab3a]
        at java.lang.Thread.run(Thread.java:1589) [?:?]
```

**Comment by fullwall** [comment]
Would you mind posting the output from this testing jar? https://www.dropbox.com/s/6teqmj1yiet41a7/Citizens-2.0.30-bUnknown.jar?dl=1

**Comment by SlimeDog** [comment]
https://blackdog.straight8.com/drops/latest.log

**Comment by SlimeDog** [comment]
BTW, in all of the failure cases, NPCs after the first five fail to load, with no WARN or ERROR notification of that fact in the log. Prior to Citizens-2770, all NPCs loaded. The save file link is listed above.

**Comment by fullwall** [comment]
https://www.dropbox.com/s/6teqmj1yiet41a7/Citizens-2.0.30-bUnknown.jar?dl=1 what about this version?

**Comment by SlimeDog** [comment]
Good. No WARN or ERROR messages, and all NPCs are loaded.

**Comment by fullwall** [comment]
Hmm, I guess one of the plugins is messing with Netty's versioning system then (deliberately I'm not sure). I'll push the change and close this out.

**Comment by SlimeDog** [comment]
Confirmed fixed with Citizens-2.0.30-b2787
- No WARN or ERROR messages related to Citizens or netty
- All NPCs load

Thanks.
